neuralmonkey/decoders/decoder.py

```diff
@@ -333,12 +336,11 @@ def step(self,
             # Run the attention mechanism.
             attns = [a.attention(cell_output) for a in att_objects]
 
-            if self._conditional_gru:
-                x_2 = linear(
-                    attns, self.embedding_size, scope="cond_gru_2_linproj")
-                # Run the RNN for the second time
-                cell_output, state = cell(
-                    x_2, state, scope="cond_gru_2_cell")
+            if self._conditional_gru and self._rnn_cell == "GRU":
+                cell_cond = self._get_conditional_gru_cell()
+                cond_input = tf.concat([cell_output] + attns, -1)
```

Comment:
I think this is not correct. When using GRU, `cell_output` is equal to `state`. To comply with the equations 11 and 12 from the GRU specification, the `cond_input` should be only the concatenation of the attentions, without the `cell_output`. The state is then provided on the next line when calling the `cell_cond`..